Add documentation for MQTT Update platform (#24636)
home-assistant/home-assistant.github.io: source/_integrations/update.mqtt.markdown

---
title: "MQTT Update"
description: "Instructions on how to interact with a device exposing an Update entity through MQTT from within Home Assistant."
ha_category:
  - Update

> ha_release: 2021.7

Reviewer: ```suggestion
ha_release: "2021.11"
```

ha_iot_class: Configurable
ha_domain: mqtt
---

> The `mqtt` Update platform allows you to integrate devices that might expose firmware/software installed and latest versions through MQTT into Home Assistant as an Update entity. Every time a message under the `topic` in the configuration is received, the entity will be updated in Home Assistant.

Reviewer: ```suggestion
The `mqtt` Update platform allows you to integrate devices that might expose firmware/software installed and the latest versions through MQTT into Home Assistant as an Update entity. Every time a message under the `topic` in the configuration is received, the entity will be updated in Home Assistant.
```

  required: false
  type: string
availability_mode:
   description: When `availability` is configured, this controls the conditions needed to set the entity to `available`. Valid entries are `all`, `any`, and `latest`. If set to `all`, `payload_available` must be received on all configured availability topics before the entity is marked as online. If set to `any`, `payload_available` must be received on at least one configured availability topic before the entity is marked as online. If set to `latest`, the last `payload_available` or `payload_not_available` received on any configured availability topic controls the availability.
   required: false
   type: string
   default: latest
availability_template:
  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract device's availability from the `availability_topic`. To determine the devices's availability result of this template will be compared to `payload_available` and `payload_not_available`."
  required: false
  type: template
command_topic:

>   description: The MQTT topic to publish command to start installing process.

Reviewer 2: May be explain what payload will be send on this topic?

Reviewer 2: Refer to `payload_install` as value that is published.

Reviewer: @bieniu Could you take a look? Or @jbouwh Could you suggest an edit?

Reviewer: (for the remaining comments, so we can get it ready before beta lands)

>   description: The MQTT topic to publish command to start installing process.

Reviewer 2: ```suggestion
  description: The MQTT topic to publish `payload_install` to start installing process.
```

  required: false
  type: string
  default: "utf-8"
entity_category:
  description: The [category](https://developers.home-assistant.io/docs/core/entity#generic-properties) of the entity.
  required: false
  type: string
  default: None
icon:
  description: "[Icon](/docs/configuration/customizing-devices/#icon) for the entity."
  required: false
  type: icon
json_attributes_template:
  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the JSON dictionary from messages received on the `json_attributes_topic`."
  required: false
  type: template
json_attributes_topic:
  description: The MQTT topic subscribed to receive a JSON dictionary payload and then set as entity attributes. Implies `force_update` of the current select state when a message is received on this topic.
  required: false
  type: string
latest_version_template:
  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the latest version value."
  required: false
  type: template
latest_version_topic:

>   description: The MQTT topic subscribed to receive update of the latest version.

Reviewer 2: ```suggestion
  description: The MQTT topic subscribed to receive an update of the latest version.
```

  required: true
  type: string
name:
  description: The name of the Select.
  required: false
  type: string
object_id:
  description: Used instead of `name` for automatic generation of `entity_id`
  required: false
  type: string
payload_install:
  description: The MQTT payload to start installing process.
  required: false
  type: string
qos:
  description: The maximum QoS level of the state topic. Default is 0 and will also be used to publishing messages.
  required: false
  type: integer
  default: 0
retain:
  description: If the published message should have the retain flag on or not.
  required: false
  type: boolean
  default: false
state_topic:

>   description: The MQTT topic subscribed to receive update of the installed version.

Reviewer 2: ```suggestion
  description: The MQTT topic subscribed to receive an update of the installed version.
```
